src/models/discriminant_analysis.jl

```diff
@@ -79,15 +87,13 @@ function _check_lda_data(model, X, y)
     p, n = size(Xm_t)
     # Recode yplain to be in {1,..., nc}
     nc == nclasses || _replace!(yplain, integers_seen, 1:nc)
-    # Check to make sure we have more than one class in training sample.
-    # This is to prevent Sb from being a zero matrix.
-    if nc <= 1
-        throw(
-            ArgumentError(
-                "The number of unique classes in "*
-                "traning sample has to be greater than one"
-            )
-        )
+
+    # issue #41
+    lone_class_unsupported = model isa Union{LDA, BayesianLDA} &&
+        !(model.cov_w == MS.SimpleCovariance() &&
```

Comment:
Although the restriction shouldn't really apply when computing `cov_w`. i.e there should be any issue in this case.
Nevertheless, I'm happy to leave things as you implemented above.

Reply:
Removed the restriction on "cov_w" and adjusted tests to make sure this works.